lib/include/tree_sitter/api.h

```diff
@@ -1091,9 +1091,11 @@ uint64_t ts_query_cursor_timeout_micros(const TSQueryCursor *self);
 /**
  * Set the range of bytes or (row, column) positions in which the query
  * will be executed.
+ *
+ * If the provided range was set, return `true`. Otherwise, return `false`.
```

Comment:
Clearer explanation

"This will return false if the start byte/point is greater than the end byte/point, otherwise it will return true.`

Also, do a favor and have the doc comments be associated to each function individually, (adjust byte/point semantics as needed). This will help w/ the rust doc comments in `bindings.rs`